src/Behaviors/Meshes/followBehavior.ts

```diff
@@ -0,0 +1,533 @@
+
+    private _angularClamp(invertView: Matrix, currentToTarget: Vector3): boolean {
+        const forward = this._tmpVectors[5];
+        forward.copyFromFloats(0, 0, 1);
+        const right = this._tmpVectors[6];
+        right.copyFromFloats(1, 0, 0);
+
+        // forward and right are related to camera frame of reference
+        Vector3.TransformNormalToRef(forward, invertView, forward);
+        Vector3.TransformNormalToRef(right, invertView, right);
+
+        // Up is global Z
+        const up = Vector3.UpReadOnly;
+
+        const dist = currentToTarget.length();
+
+        if (dist < Epsilon) {
+            return false;
+        }
+
+        let angularClamped = false;
+        const rotationQuat = this._tmpQuaternion;
+
+        // X-axis leashing
+        if (this.ignoreCameraPitchAndRoll) {
+            const angle = this._angleBetweenOnPlane(currentToTarget, forward, right);
+            Quaternion.RotationAxisToRef(right, angle, rotationQuat);
+            currentToTarget.rotateByQuaternionToRef(rotationQuat, currentToTarget);
+        } else {
+            const angle = -this._angleBetweenOnPlane(currentToTarget, forward, right);
+            const minMaxAngle = ((this.maxViewVerticalDegrees * Math.PI) / 180) * 0.5;
+            if (angle < -minMaxAngle) {
+                Quaternion.RotationAxisToRef(right, -angle - minMaxAngle, rotationQuat);
+                currentToTarget.rotateByQuaternionToRef(rotationQuat, currentToTarget);
+                angularClamped = true;
+            } else if (angle > minMaxAngle) {
+                Quaternion.RotationAxisToRef(right, -angle + minMaxAngle, rotationQuat);
+                currentToTarget.rotateByQuaternionToRef(rotationQuat, currentToTarget);
+                angularClamped = true;
+            }
+        }
+
+        // Y-axis leashing
+        const angle = this._angleBetweenVectorAndPlane(currentToTarget, right);
+        const minMaxAngle = ((this.maxViewHorizontalDegrees * Math.PI) / 180) * 0.5;
+        if (angle < -minMaxAngle) {
+            Quaternion.RotationAxisToRef(up, -angle - minMaxAngle, rotationQuat);
+            currentToTarget.rotateByQuaternionToRef(rotationQuat, currentToTarget);
+            angularClamped = true;
+        } else if (angle > minMaxAngle) {
+            Quaternion.RotationAxisToRef(up, -angle + minMaxAngle, rotationQuat);
+            currentToTarget.rotateByQuaternionToRef(rotationQuat, currentToTarget);
+            angularClamped = true;
+        }
+
+        return angularClamped;
+    }
+
+    private _orientationClamp(currentToTarget: Vector3, rotationQuaternion: Quaternion) {
+        // Construct a rotation quat from up vector and target vector
+        const toFollowed = this._tmpVectors[0];
+        toFollowed.copyFrom(currentToTarget).scaleInPlace(-1).normalize();
+
+        const up = this._tmpVectors[1];
+        const right = this._tmpVectors[2];
+        // We use global up vector to orient the following node (global +Y)

```

Comment:
I agree that hard-coding up vector as +Y is a big assumption, but in case of head-mounted displays, up vector can change quite dramatically with the pitch/roll of the head, while we still want slates to be gravity-aligned. 
Are you aware of a less camera-dependant way of retrieving the scene's up axis ?

NB: Now that I think about it, your comment is all the more relevant as I'm also hard coding forward axii as +Z when this is only valid with left-handed conventions.

Reply:
So I ended up keeping the assumption that gravity is aligned with Y axis (which is anyway always the case in XR) but I made the follow behavior taking `scene.useRightHandedSystem` into account.